app/lib/api_authentication/by_access_token.rb

```diff
@@ -108,8 +108,8 @@ def authenticated_token
       @authenticated_token = domain_account.access_tokens.find_from_value(access_token) if access_token
     end
 
-    def enforce_access_token_permission
-      PermissionEnforcer.enforce(authenticated_token, &Proc.new)
+    def enforce_access_token_permission(&block)
+      PermissionEnforcer.enforce(authenticated_token, &block)
```

Comment:
wtf?! I see this `&Proc.new` for the first time and it looks like a horrible hack.

Described here:
https://stackoverflow.com/a/49870776/520567
https://mudge.name/2011/01/26/passing-blocks-in-ruby-without-block/

I wonder if we better use this syntax:
```suggestion
    def enforce_access_token_permission(...)
      PermissionEnforcer.enforce(authenticated_token, ...)
```

Reply:
Yeah I also find `&Proc.new` extremely confusing :dizzy_face: 

I am not sure about replacing`&block` with `...` though - `enforce_access_token_permission` is a method that is called from the `around_action` hook, so I understand that by definition it will only be able to accept a block, not any other arguments, so I'd prefer to keep only the block.